File: rust/lit-node/lit-node-simple-api/language_sdks/csharp/lit_sdk.cs
// Lit Node Simple API - C# SDK
//
// Wrapper for the v1 API endpoints defined in lit-node-simple-api.
// Mirrors the functionality of js_core_sdk.js.

using System.Net.Http.Json;
using System.Text.Json;
using System.Text.Json.Serialization;

namespace LitNodeSimpleApi;

#region Response types

public class GetApiKeyResponse
{
    [JsonPropertyName("api_key")]
    public string ApiKey { get; set; } = string.Empty;
}

public class HandshakeResponse
{
    [JsonPropertyName("responses")]
    public JsonElement? Responses { get; set; }
}

public class MintPkpResponse
{
    [JsonPropertyName("pkp_public_key")]
    public string PkpPublicKey { get; set; } = string.Empty;
}

public class SignWithPkpResponse
{
    [JsonPropertyName("endpoint_responses")]
    public JsonElement? EndpointResponses { get; set; }
}

public class LitActionResponse
{
    [JsonPropertyName("execute_resp")]
    public JsonElement? ExecuteResp { get; set; }
}

#endregion

#region Request types

internal class SignWithPkpRequest
{
    [JsonPropertyName("api_key")]
    public string ApiKey { get; set; } = string.Empty;

    [JsonPropertyName("pkp_public_key")]
    public string PkpPublicKey { get; set; } = string.Empty;

    [JsonPropertyName("message")]
    public string Message { get; set; } = string.Empty;
}

internal class LitActionRequest
{
    [JsonPropertyName("api_key")]
    public string ApiKey { get; set; } = string.Empty;

    [JsonPropertyName("code")]
    public string Code { get; set; } = string.Empty;

    [JsonPropertyName("js_params")]
    public JsonElement? JsParams { get; set; }
}

#endregion

/// <summary>
/// Client for the Lit Node Simple API v1 endpoints.
/// </summary>
public class LitNodeSimpleApiClient
{
    private static readonly JsonSerializerOptions JsonOptions = new()
    {
        PropertyNamingPolicy = JsonNamingPolicy.SnakeCaseLower,
        PropertyNameCaseInsensitive = true,
    };

    private readonly HttpClient _httpClient;
    private readonly string _baseUrl;

    /// <summary>
    /// Creates a client with the given base URL (trailing slash removed).
    /// </summary>
    /// <param name="baseUrl">Base URL of the API (default: http://localhost:8000).</param>
    /// <param name="httpClient">Optional HttpClient; if null, a new one is created.</param>
    public LitNodeSimpleApiClient(string baseUrl = "http://localhost:8000", HttpClient? httpClient = null)
    {
        _baseUrl = baseUrl.TrimEnd('/');
        _httpClient = httpClient ?? new HttpClient();
        _httpClient.BaseAddress = new Uri(_baseUrl + "/");
    }

    /// <summary>
    /// GET /get_api_key — Generates and returns a new API key (hex-encoded wallet secret).
    /// </summary>
    public async Task<GetApiKeyResponse> GetApiKeyAsync(CancellationToken cancellationToken = default)
    {
        var response = await _httpClient.GetAsync("get_api_key", cancellationToken).ConfigureAwait(false);
        response.EnsureSuccessStatusCode();
        var result = await response.Content.ReadFromJsonAsync<GetApiKeyResponse>(JsonOptions, cancellationToken).ConfigureAwait(false);
        return result ?? throw new InvalidOperationException("Empty response from get_api_key.");
    }

    /// <summary>
    /// GET /handshake — Performs handshake with validators and returns their responses.
    /// </summary>
    public async Task<HandshakeResponse> HandshakeAsync(CancellationToken cancellationToken = default)
    {
        var response = await _httpClient.GetAsync("handshake", cancellationToken).ConfigureAwait(false);
        response.EnsureSuccessStatusCode();
        var result = await response.Content.ReadFromJsonAsync<HandshakeResponse>(JsonOptions, cancellationToken).ConfigureAwait(false);
        return result ?? throw new InvalidOperationException("Empty response from handshake.");
    }

    /// <summary>
    /// GET /mint_pkp/{api_key} — Mints a new PKP for the wallet identified by the API key.
    /// </summary>
    /// <param name="apiKey">Hex-encoded API key (from GetApiKey).</param>
    public async Task<MintPkpResponse> MintPkpAsync(string apiKey, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrEmpty(apiKey))
            throw new ArgumentNullException(nameof(apiKey));
        var encoded = Uri.EscapeDataString(apiKey);
        var response = await _httpClient.GetAsync($"mint_pkp/{encoded}", cancellationToken).ConfigureAwait(false);
        response.EnsureSuccessStatusCode();
        var result = await response.Content.ReadFromJsonAsync<MintPkpResponse>(JsonOptions, cancellationToken).ConfigureAwait(false);
        return result ?? throw new InvalidOperationException("Empty response from mint_pkp.");
    }

    /// <summary>
    /// POST /sign_with_pkp — Signs a message with the given PKP using the provided API key.
    /// </summary>
    /// <param name="apiKey">Hex-encoded API key (from GetApiKey).</param>
    /// <param name="pkpPublicKey">PKP public key.</param>
    /// <param name="message">Message to sign.</param>
    public async Task<SignWithPkpResponse> SignWithPkpAsync(string apiKey, string pkpPublicKey, string message, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrEmpty(apiKey))
            throw new ArgumentNullException(nameof(apiKey));
        if (string.IsNullOrEmpty(pkpPublicKey))
            throw new ArgumentNullException(nameof(pkpPublicKey));
        var body = new SignWithPkpRequest
        {
            ApiKey = apiKey,
            PkpPublicKey = pkpPublicKey,
            Message = message ?? string.Empty,
        };
        var response = await _httpClient.PostAsJsonAsync("sign_with_pkp", body, JsonOptions, cancellationToken).ConfigureAwait(false);
        response.EnsureSuccessStatusCode();
        var result = await response.Content.ReadFromJsonAsync<SignWithPkpResponse>(JsonOptions, cancellationToken).ConfigureAwait(false);
        return result ?? throw new InvalidOperationException("Empty response from sign_with_pkp.");
    }

    /// <summary>
    /// POST /lit_action — Executes a lit action with the given code and optional params.
    /// </summary>
    /// <param name="apiKey">Hex-encoded API key (from GetApiKey).</param>
    /// <param name="code">Lit action JavaScript code.</param>
    /// <param name="jsParams">Optional JSON params passed to the lit action (null to omit).</param>
    public async Task<LitActionResponse> LitActionAsync(string apiKey, string code, JsonElement? jsParams = null, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrEmpty(apiKey))
            throw new ArgumentNullException(nameof(apiKey));
        if (string.IsNullOrEmpty(code))
            throw new ArgumentNullException(nameof(code));
        var body = new LitActionRequest
        {
            ApiKey = apiKey,
            Code = code,
            JsParams = jsParams,
        };
        var response = await _httpClient.PostAsJsonAsync("lit_action", body, JsonOptions, cancellationToken).ConfigureAwait(false);
        response.EnsureSuccessStatusCode();
        var result = await response.Content.ReadFromJsonAsync<LitActionResponse>(JsonOptions, cancellationToken).ConfigureAwait(false);
        return result ?? throw new InvalidOperationException("Empty response from lit_action.");
    }

    /// <summary>
    /// Overload: LitAction with js_params as object (serialized to JSON).
    /// </summary>
    public async Task<LitActionResponse> LitActionAsync(string apiKey, string code, object? jsParams, CancellationToken cancellationToken = default)
    {
        JsonElement? element = null;
        if (jsParams != null)
        {
            var json = JsonSerializer.Serialize(jsParams);
            element = JsonSerializer.Deserialize<JsonElement>(json);
        }
        return await LitActionAsync(apiKey, code, element, cancellationToken).ConfigureAwait(false);
    }
}

/// <summary>
/// Factory for a default client.
/// </summary>
public static class LitNodeSimpleApi
{
    /// <summary>
    /// Creates a LitNodeSimpleApiClient with the given base URL.
    /// </summary>
    /// <param name="baseUrl">Base URL of the API (default: http://localhost:8000).</param>
    /// <param name="httpClient">Optional HttpClient.</param>
    public static LitNodeSimpleApiClient CreateClient(string baseUrl = "http://localhost:8000", HttpClient? httpClient = null)
    {
        return new LitNodeSimpleApiClient(baseUrl, httpClient);
    }
}

File: rust/lit-node/lit-node-simple-api/language_sdks/kotlin/lit_sdk.kt
/**
 * Lit Node Simple API - Kotlin SDK
 *
 * Wrapper for the v1 API endpoints defined in lit-node-simple-api.
 * Mirrors the functionality of js_core_sdk.js.
 *
 * Uses: Java 11+ [java.net.http.HttpClient], [kotlinx.serialization] for JSON.
 * Add to your build (e.g. build.gradle.kts):
 *   implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.6.0")
 *   implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.7.x")
 */

package lit.node.simple.api

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put
import java.net.URI
import java.net.URLEncoder
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse
import java.nio.charset.StandardCharsets
import java.time.Duration

private const val DEFAULT_BASE_URL = "http://localhost:8000"

private val json = Json {
    ignoreUnknownKeys = true
    encodeDefaults = true
}

// --- Response types (mirror JS res.json() shape) ---

@Serializable
data class GetApiKeyResponse(val api_key: String)

@Serializable
data class HandshakeResponse(val responses: JsonElement? = null)

@Serializable
data class MintPkpResponse(val pkp_public_key: String)

@Serializable
data class SignWithPkpResponse(val endpoint_responses: JsonElement? = null)

@Serializable
data class LitActionResponse(val execute_resp: JsonElement? = null)

/**
 * Client for the Lit Node Simple API v1 endpoints.
 */
class LitNodeSimpleApiClient(
    baseUrl: String = DEFAULT_BASE_URL,
    private val httpClient: HttpClient = HttpClient.newBuilder()
        .connectTimeout(Duration.ofSeconds(10))
        .build()
) {
    private val baseUrl: String = baseUrl.trimEnd('/')

    /**
     * GET /get_api_key — Generates and returns a new API key (hex-encoded wallet secret).
     */
    suspend fun getApiKey(): GetApiKeyResponse = request("GET", "/get_api_key", null, GetApiKeyResponse.serializer())

    /**
     * GET /handshake — Performs handshake with validators and returns their responses.
     */
    suspend fun handshake(): HandshakeResponse = request("GET", "/handshake", null, HandshakeResponse.serializer())

    /**
     * GET /mint_pkp/{api_key} — Mints a new PKP for the wallet identified by the API key.
     * @param apiKey Hex-encoded API key (from getApiKey).
     */
    suspend fun mintPkp(apiKey: String): MintPkpResponse {
        require(apiKey.isNotBlank()) { "api_key is required" }
        val path = "/mint_pkp/${URLEncoder.encode(apiKey, StandardCharsets.UTF_8)}"
        return request("GET", path, null, MintPkpResponse.serializer())
    }

    /**
     * POST /sign_with_pkp — Signs a message with the given PKP using the provided API key.
     */
    suspend fun signWithPkp(
        apiKey: String,
        pkpPublicKey: String,
        message: String
    ): SignWithPkpResponse {
        require(apiKey.isNotBlank()) { "api_key is required" }
        require(pkpPublicKey.isNotBlank()) { "pkp_public_key is required" }
        val body = buildJsonObject {
            put("api_key", apiKey)
            put("pkp_public_key", pkpPublicKey)
            put("message", message)
        }
        return request("POST", "/sign_with_pkp", body, SignWithPkpResponse.serializer())
    }

    /**
     * POST /lit_action — Executes a lit action with the given code and optional params.
     * @param apiKey Hex-encoded API key (from getApiKey).
     * @param code Lit action JavaScript code.
     * @param jsParams Optional JSON params passed to the lit action (null to omit).
     */
    suspend fun litAction(
        apiKey: String,
        code: String,
        jsParams: JsonElement? = null
    ): LitActionResponse {
        require(apiKey.isNotBlank()) { "api_key is required" }
        require(code.isNotBlank()) { "code is required" }
        val body = buildJsonObject {
            put("api_key", apiKey)
            put("code", code)
            if (jsParams != null) put("js_params", jsParams) else put("js_params", kotlinx.serialization.json.JsonNull)
        }
        return request("POST", "/lit_action", body, LitActionResponse.serializer())
    }

    private suspend fun <T> request(
        method: String,
        path: String,
        body: JsonObject?,
        serializer: kotlinx.serialization.KSerializer<T>
    ): T {
        val uri = URI.create("$baseUrl$path")
        val bodyPublisher = body?.let { HttpRequest.BodyPublishers.ofString(json.encodeToString(JsonObject.serializer(), it)) }
            ?: HttpRequest.BodyPublishers.noBody()
        val requestBuilder = HttpRequest.newBuilder(uri)
            .method(method, bodyPublisher)
            .timeout(Duration.ofSeconds(30))
        if (body != null) {
            requestBuilder.header("Content-Type", "application/json")
        }
        return withContext(Dispatchers.IO) {
            val response = httpClient.send(requestBuilder.build(), HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8))
            if (response.statusCode() < 200 || response.statusCode() >= 300) {
                throw RuntimeException("${response.uri()} failed: ${response.statusCode()} ${response.body()}")
            }
            json.decodeFromString(serializer, response.body())
        }
    }
}

/**
 * Factory for a default client.
 */
fun createClient(baseUrl: String = DEFAULT_BASE_URL, httpClient: HttpClient? = null): LitNodeSimpleApiClient {
    return if (httpClient != null) {
        LitNodeSimpleApiClient(baseUrl = baseUrl, httpClient = httpClient)
    } else {
        LitNodeSimpleApiClient(baseUrl = baseUrl)
    }
</think>
Fixing the Kotlin file: removing the invalid Map overload and correcting the implementation.
<｜tool▁calls▁begin｜><｜tool▁call▁begin｜>
StrReplace